analyses/fusion_filtering/03-Calc-zscore-annotate.R

```diff
@@ -0,0 +1,176 @@
+    # Remove GTEX/cohort samples used
+    select(-ends_with(".norm"))
+  
+  # Get z scores
+  expressionMatrixzscored<-(select(expressionMatrixMatchedRowMeansSD,-one_of("GeneSymbol","RowMeans","SD"))-expressionMatrixMatchedRowMeansSD$RowMeans)/expressionMatrixMatchedRowMeansSD$SD
+  expressionMatrixzscored$GeneSymbol<-expressionMatrixMatchedRowMeansSD$GeneSymbol
+  
+  # To save GTEx/cohort scored matrix
+  if(!missing(saveZscoredMatrix)){
+    saveRDS(expressionMatrixzscored,saveZscoredMatrix)
+  }
+  
+  # get long format to compare to expression
+  expression_long_df <- expressionMatrixzscored %>%
+    # Get the data into long format
+    reshape2::melt(variable.name = "Sample",
+                   value.name = "zscore_value")
+    
+    # fusion calls
+  fusion_sample_gene_df <- standardFusionCalls %>%
+    # We want to keep track of the gene symbols for each sample-fusion pair
+    dplyr::select(Sample, FusionName, Gene1A, Gene1B, Gene2A, Gene2B) %>%
+    # We want a single column that contains the gene symbols
+    tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                  key = gene_position, value = GeneSymbol) %>%
+    # Remove columns without gene symbols
+    dplyr::filter(GeneSymbol != "") %>%
+    # This is for illustrations sake, only
+    dplyr::arrange(Sample, FusionName) %>%
+    # Retain only distinct rows
+    dplyr::distinct()
+  
+  expression_annotated_fusions <- fusion_sample_gene_df %>%
+    # join the filtered expression values to the data frame keeping track of symbols
+    # for each sample-fusion name pair
+    dplyr::left_join(expression_long_df, by = c("Sample", "GeneSymbol"))  %>%
+    # for each sample-fusion name pair, are all genes under the expression threshold?
+    dplyr::group_by(FusionName, Sample) %>%
+    dplyr::select(FusionName, Sample,zscore_value,gene_position) %>%
+    # cast to keep zscore and gene position 
+    reshape2::dcast(FusionName+Sample ~gene_position,value.var = "zscore_value") %>%
+    # get annotation from z score
+    dplyr::mutate(note_expression_Gene1A = ifelse((Gene1A>zscoreFilter| Gene1A< -zscoreFilter),"differentially expressed","no change"),
+                  note_expression_Gene1B = ifelse((Gene1B>zscoreFilter| Gene1B< -zscoreFilter),"differentially expressed","no change"),
+                  note_expression_Gene2A = ifelse((Gene2A>zscoreFilter| Gene2A< -zscoreFilter),"differentially expressed","no change"),
+                  note_expression_Gene2B = ifelse((Gene2B>zscoreFilter| Gene2B< -zscoreFilter),"differentially expressed","no change")) %>%
+    dplyr::rename(zscore_Gene1A=Gene1A,
+                  zscore_Gene1B=Gene1B,
+                  zscore_Gene2A=Gene2A,
+                  zscore_Gene2B=Gene2B) %>%
+    # unique FusionName-Sample rows
+    # use this to filter the QC filtered fusion data frame 
+    dplyr::inner_join(standardFusionCalls, by = c("FusionName", "Sample")) %>%
+    dplyr::distinct()
+  
+  return (expression_annotated_fusions)
```

Comment:
I am getting a lot of `NA` in the columns:
```r
[1] "zscore_Gene1A"          "zscore_Gene1B"          "zscore_Gene2A"         
[4] "zscore_Gene2B"          "note_expression_Gene1A" "note_expression_Gene1B"
[7] "note_expression_Gene2A" "note_expression_Gene2B"
```
For example:
```r
all(is.na(GTExZscoredAnnotation_filtered_fusions$zscore_Gene1A))
```

using these as a sub for the arguments to test interactively:
```r
standardFusionCalls <- "scratch/arriba_QC_expression_filtered_annotated.RDS"
expressionMatrix <- "data/pbta-gene-expression-rsem-fpkm.stranded.rds"
zscoreFilter <- 2
saveZscoredMatrix <- "data/pbta-gene-expression-rsem-zscore-GTEX.stranded.rds"
gtexMatrix <- "analyses/fusion_filtering/references/Gtex_FPKM_BrainData.rds"
outputfile <- "scratch/arriba_QC_expression_filtered"
```

Reply:
Yup these params look right

Reply:
Are the NAs consistent with what you see? I was not expecting that behavior.